src/LibraryManager.Vsix/Shared/VsHelpers.cs

```diff
@@ -222,6 +226,38 @@ public static bool IsKind(this Project project, params string[] kindGuids)
             return false;
         }
 
+        public static async Task<bool> DeleteFilesFromProjectAsync (Project project, IEnumerable<string> filePaths, Action<string, LogLevel> logAction, CancellationToken cancellationToken)
+        {
+            if (IsCapabilityMatch(project, Constants.DotNetCoreWebCapability))
+            {
+                return false;
+            }
+
+            int batchSize = 10;
+
+            IVsHierarchy hierarchy = GetHierarchy(project);
+            IVsProjectBuildSystem bldSystem = hierarchy as IVsProjectBuildSystem;
+            List<string> filesToRemove = filePaths.ToList();
+
+            while (filesToRemove.Any())
+            {
+                List<string> nextBatch = filesToRemove.Take(batchSize).ToList();
+                bool success = await DeleteProjectItemsInBatchAsync(hierarchy, nextBatch, logAction, cancellationToken);
+
+                if(!success)
+                {
+                    return false;
```

Comment:
should we have a retry?  I don't know if it's worth it, depends on if DeleteProjectItemsInBatchAsync is flaky I guess.